hub/router/secrets.go

```diff
@@ -0,0 +1,350 @@
+package router
+
+import (
+	"GADS/common/db"
+	"GADS/hub/auth"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// GetSecretKeys returns all secret keys
+func GetSecretKeys(c *gin.Context) {
+	// Get secret keys from database
+	store := auth.NewSecretStore(db.GlobalMongoStore.GetDefaultDatabase())
+	keys, err := store.GetAllSecretKeys()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get secret keys"})
+		return
+	}
+
+	// Don't expose the actual secret key values in the response for security
+	var response []gin.H = []gin.H{}
+	for _, key := range keys {
+		response = append(response, gin.H{
+			"id":                      key.ID.Hex(),
+			"origin":                  key.Origin,
+			"is_default":              key.IsDefault,
+			"created_at":              key.CreatedAt,
+			"updated_at":              key.UpdatedAt,
+			"user_identifier_claim":   key.UserIdentifierClaim,
+			"tenant_identifier_claim": key.TenantIdentifierClaim,
+		})
+	}
+
+	c.JSON(http.StatusOK, gin.H{"secret_keys": response})
+}
+
+// AddSecretKey adds a new secret key
+func AddSecretKey(c *gin.Context) {
+	// Get username from user claims for audit
+	username, exists := getUsernameFromContext(c)
+	if !exists {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get user information"})
+		return
+	}
+
+	// Parse request body
+	var request struct {
```

Comment:
Can't we reuse the `SecretKey` struct here and update the necessary fields after before adding it in the DB?

Reply:
It’s not recommended to use a database model to parse the request body, as the user might modify fields that shouldn’t be changed, like the ID.